src/common/src/config/merge.rs

```diff
@@ -188,27 +188,53 @@ mod tests {
         let partial = r#"
             [streaming]
             no_such_entry = 114514
+
+            [streaming.developer]
+            no_such_dev_entry = 1919810
         "#;
-        let merged = merge_config(&base, partial, ["streaming"])
+        let merged = merge_streaming_config_section(&base, partial)
             .unwrap()
             .unwrap();
 
         let unrecognized = merged.unrecognized.into_inner();
         assert_eq!(unrecognized.len(), 1);
         assert_eq!(unrecognized["no_such_entry"], 114514);
+
+        let dev_unrecognized = merged.developer.unrecognized.into_inner();
+        assert_eq!(dev_unrecognized.len(), 1);
+        assert_eq!(dev_unrecognized["no_such_dev_entry"], 1919810);
     }
 
     #[test]
     fn test_invalid_type() {
         let base = StreamingConfig::default();
         let partial = r#"
             [streaming.developer]
-            stream_chunk_size = "omakase"
+            chunk_size = "omakase"
         "#;
-        let error = merge_config(&base, partial, ["streaming"]).unwrap_err();
+        let error = merge_streaming_config_section(&base, partial).unwrap_err();
         expect_test::expect![[r#"
             failed to deserialize merged config: invalid type: string "omakase", expected usize
-            in `developer.stream_chunk_size`
+            in `developer.chunk_size`
+        "#]]
+        .assert_eq(&error.to_report_string());
+    }
+
+    // Even though we accept `stream_` prefixed config key when deserializing the config, since
+    // we perform merging atop of the raw `toml::Value`, we don't have the information about
+    // the aliasing. Therefore, using a prefixed config key in config override will result in
+    // a duplicate field error.
```

Comment:
This is okay as per-job configuration is a new feature and there's no backward compatibility issues. We just need to update the documentation and instruct users to specify config keys without the prefix.